source/core/NumericString.cpp

```diff
@@ -2027,8 +2027,13 @@ Boolean DateTimeToString(const Date& date, Boolean isUTC, SubString* format, Str
     while (validFormatString && tempFormat.ReadRawChar(&c)) {
         if (c == '%') {
             TimeFormatOptions fOption;
+            FormatOptions fFormatOption;
+
+            Boolean parseFinished = false;
+            if (ReadLocalizedDecimalSeparator(&tempFormat, 0, null, null, &tempFormat, &validFormatString, &fFormatOption, &parseFinished))
+                continue;
```

Comment:
FormatOptions is a big structure with lots of fields that don't apply to time formats.
Could you do a small refactor of ReadLocalizedDecimalSeparator to make it take the address of a 'char' decimalSeparator variable only, rather than the structure, and make Format() call it with &fOption.decimalSeparator, and this place call it with the address of a char local?

Also, modify the %u case to actually use the decimal separator value below.

Reply:
Do we need to also consider the %U case?

Reply:
No, %U is week number, only %u deals with fractional numbers with a decimal point.